src/data-table/data-table.tsx

```diff
@@ -28,27 +34,72 @@ export const DataTable = ({
     loadState = "success",
     enableMultiSelect,
     selectedIds,
+    disabledIds,
     enableSelectAll,
+    enableActionBar,
     emptyView,
+    actionBarContent,
     renderCustomEmptyView,
     onHeaderClick,
     onSelect,
     onSelectAll,
+    onClearSelectionClick,
     ...otherProps
 }: DataTableProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const [showLastBorder, setShowLastBorder] = useState(false);
+    const [isFloating, setIsFloating] = useState(false);
+    const [alignActionBarWithScreen, setAlignActionBarWithScreen] =
+        useState(false);
+    const tableRef = useRef<HTMLTableElement>();
+
+    const { ref: stickyRef, inView: isLastRowInView } = useInView({
+        threshold: 0,
+    });
+    const { ref: endRef, inView: atEnd } = useInView({
+        threshold: 0,
+    });
+    const { ref: containerRef, inView: isContainerInView } = useInView({
+        threshold: 0.2,
```

Comment:
this threshold is percentage based so it might not work when the table does not have a fixed height or when it's very tall relative to the viewport

an alternative is using root margin to adjust the offset
```
threshold: 0,
rootMargin: "0px 0px -100px 0px", <--- adjust based on the header height
```

but note that this doesn't work in the Canvas tab due to iframe limitations in Storybook